client/src/app/shared/validators/one-of-validator.ts

```diff
@@ -0,0 +1,30 @@
+import { ValidationErrors, AbstractControl } from '@angular/forms';
+
+/**
+ * Custom validator class
+ */
+export class OneOfValidator {
+    /**
+     * Function to confirm that at least one of the given form controls has a valid value.
+     *
+     * @param keys Undefined amount of form control names.
+     *
+     * @returns An error if all of these form controls have no valid values or null if at least one is filled.
+     */
+    public static validation(...keys: string[]): ValidationErrors | null {
```

Comment:
Return type is `(control: AbstractControl) => ValidationErrors | null`. I'm not sure why the linter doesn't noticed this.